plugins/library/messagehub.py

```diff
@@ -0,0 +1,161 @@
+    returned: changed
+    type: dictionary
+    sample: {"messages": [...]}
+'''
+
+
+import stomp
+import logging
+import logging.config
+import time
+import os
+import json
+from ansible.module_utils.basic import AnsibleModule
+
+
+def setup_logging(default_path='logging.json', default_level=logging.INFO, env_key='LOG_CFG'):
+    """Setup logging configuration
+    :param default_path Default path to logging.json file
+    :param default_level Level of logging
+    :param env_key Environmental variable which contains logging configuration file path
+
+    """
+    path = default_path
+    value = os.getenv(env_key, None)
+    if value:
+        path = value
+    if os.path.exists(path):
+        with open(path, 'rt') as f:
+            config = json.load(f)
+        logging.config.dictConfig(config)
+    else:
+        logging.basicConfig(level=default_level)
+
+
+class CIListener(stomp.ConnectionListener):
+    def __init__(self, count):
+        self.count = count
+        self.metamorph_data = []
+        self.error_message = {}
+
+    def on_error(self, headers, message):
+        self.error_message['headers'] = headers
+        self.error_message['message'] = message
+
+    def on_message(self, headers, message):
+        if self.count > len(self.metamorph_data):
+            self.metamorph_data.append({"header": headers, "message": message})
+            self.count += 1
+
+
+def main():
+    fields = {
+        "user": {"required": True, "type": "str"},
+        "password": {"required": True, "type": "str"},
+        "selector": {"required": False, "type": "str"},
+        "host": {"required": True, "type": "str"},
+        "port": {"default": 61613, "type": "int"},
+        "destination": {"default": '/topic/CI', "type": "str"},
+        "count": {"default": 1, "type": "int"}
+    }
+    setup_logging(default_path="../../logging.json")
+    module = AnsibleModule(argument_spec=fields)
+    conn = stomp.Connection([(module.params['host'], module.params['port'])])
+    listener = CIListener(module.params['count'])
+    conn.set_listener('CI Listener', listener)
+    conn.start()
+    conn.connect(login=module.params['user'], passcode=module.params['password'], wait=True)
+    if module.params['selector']:
+        conn.subscribe(
+            destination=module.params['destination'],
+            id='1',
+            ack='auto',
+            headers={'selector': module.params['selector']}
+        )
+    else:
+        conn.subscribe(
+            destination=module.params['destination'],
+            id='1',
+            ack='auto'
+        )
+    logging.info("Connection to message bus established.")
+    while (not listener.error_message) and len(listener.metamorph_data) < module.params['count']:
+        logging.info("Waiting 1s for CI message to arrive")
+        time.sleep(1)
+    conn.disconnect()
+
+    if not listener.error_message:
+        module.exit_json(changed=True, meta=dict(messages=listener.metamorph_data))
```

Comment:
Can we have listener.metamorph_data dumped to metamorph.json along with module.exit_json ?